src/main/default/aura/Lookup/Lookup.cmp

```diff
@@ -1,79 +1,104 @@
-                            <div class="{! 'slds-combobox__form-element slds-input-has-icon '+ (empty(v.selection) ? 'slds-input-has-icon_right' : 'slds-input-has-icon_left-right') }" role="none">
-                                <lightning:icon iconName="{! empty(v.selection[0].icon) ? 'standard:default' : v.selection[0].icon}" size="small" alternativeText="Selected item icon"
-                                    class="{! 'slds-combobox__input-entity-icon '+ (empty(v.selection) ? 'slds-hide' : '') }"/>
-                                <input type="text" class="slds-input slds-combobox__input slds-combobox__input-value" aria-autocomplete="list" aria-controls="{# globalId + '_listbox' }" autocomplete="off" role="textbox"
-                                    id="{# globalId + '_combobox' }"
-                                    aura:id="searchInput"
-                                    placeholder="{!v.placeholder}"
-                                    value="{! empty(v.selection) ? '' : v.selection[0].title }"
-                                    onfocus="{!c.onFocus}"
-                                    onblur="{!c.onBlur}"
-                                    oninput="{!c.onInput}"
-                                    readonly="{! !empty(v.selection) }"/>
-                                <lightning:icon iconName="utility:search" size="x-small" alternativeText="Search icon"
-                                    class="{! 'slds-input__icon slds-input__icon_right '+ (empty(v.selection) ? '' : 'slds-hide') }"/>
+                            <div class="{! 'slds-combobox__form-element slds-input-has-icon '+ (empty(v.selection) ? 'slds-input-has-icon_right' : 'slds-input-has-icon_left-right') }"
+                                role="none">
+                                <lightning:icon iconName="{! empty(v.selection[0].icon) ? 'standard:default' : v.selection[0].icon}"
+                                    size="small" alternativeText="Selected item icon" class="{! 'slds-combobox__input-entity-icon '+ (empty(v.selection) ? 'slds-hide' : '') }" />
+                                <input type="text" class="{! 'slds-input slds-combobox__input slds-combobox__input-value has-custom-height ' + (!empty(v.errors) ? 'has-custom-error' : '') + (!empty(v.selection) ? ' has-custom-border' : '') }"
+                                    aria-autocomplete="list" aria-controls="{# globalId + '_listbox' }" autocomplete="off"
+                                    role="textbox" id="{# globalId + '_combobox' }" aura:id="searchInput" placeholder="{!v.placeholder}"
+                                    value="{! empty(v.selection) ? '' : v.selection[0].title }" onfocus="{!c.onFocus}"
+                                    onblur="{!c.onBlur}" oninput="{!c.onInput}" readonly="{! !empty(v.selection) }" />
+
+                                <!-- search icon and spinner -->
+                                <aura:if isTrue="{!v.isLoading}">
```

Comment:
Same comment as earlier, please switch to dynamic CSS classes instead of `aura:if`.